### PR TITLE
Adds config options for randomly flipping images  

### PR DESCRIPTION
**Pull request by kondela**
Adds options to config in order to enable/disable random horizontal/vertical flipping for image augmentation.

Feature labeled as _enhancement_ and requested by [1513](https://github.com/facebookresearch/detectron2/issues/1513)

### CODE REVIEW

**Comment by kondela** [comment]
Pipeline failed on `test_nms_rotated_0_degree_cuda` located in `tests/layers/test_nms_rotated.py`. However, when I ran it locally it passed all the tests. After a closer inspection it appears that the `_create_tensors` method is non deterministic as it contains `torch.rand` which once in a while produces values that make the test fail. Confirmed after running the test 200 times it failed once. 

I guess this would deserve a new PR that would make test behavior deterministic. Meanwhile, I propose to simply rerun the failed pipeline, what do you think @ppwwyyxx?

File: detectron2/config/defaults.py [hunk 1]
```diff
@@ -57,6 +57,10 @@
 _C.INPUT.MIN_SIZE_TEST = 800
 # Maximum size of the side of the image during testing
 _C.INPUT.MAX_SIZE_TEST = 1333
+# `True` if flipping is used for data augmentation during training
+_C.INPUT.RANDOM_FLIP = CN({"ENABLED": True})
+_C.INPUT.RANDOM_FLIP.HORIZONTAL = True
+_C.INPUT.RANDOM_FLIP.VERTICAL = False
```
Comment (ppwwyyxx) [inline detectron2/config/defaults.py:63]: could you use `INPUT.RANDOM_FLIP="horizontal/vertical/none"`?

**Review by ppwwyyxx — COMMENTED** [review]
Yeah this test occasionally fails. Instead of making it deterministic we would actually like to understand the root cause of that and whether there is a small bug behind. For now we can just ignore that in this PR

File: detectron2/config/defaults.py [hunk 2]
```diff
@@ -58,9 +58,7 @@
 # Maximum size of the side of the image during testing
 _C.INPUT.MAX_SIZE_TEST = 1333
 # `True` if flipping is used for data augmentation during training
-_C.INPUT.RANDOM_FLIP = CN({"ENABLED": True})
-_C.INPUT.RANDOM_FLIP.HORIZONTAL = True
-_C.INPUT.RANDOM_FLIP.VERTICAL = False
+_C.INPUT.RANDOM_FLIP = "horizontal/vertical/none"
```
Comment (ppwwyyxx) [inline detectron2/config/defaults.py:61]: default should be "horizontal"; comment should explain possible values

File: detectron2/data/detection_utils.py [hunk 3]
```diff
@@ -579,8 +579,13 @@ def build_augmentation(cfg, is_train):
         max_size = cfg.INPUT.MAX_SIZE_TEST
         sample_style = "choice"
     augmentation = [T.ResizeShortestEdge(min_size, max_size, sample_style)]
-    if is_train:
-        augmentation.append(T.RandomFlip())
+    if is_train and "none" not in cfg.INPUT.RANDOM_FLIP:
```
Comment (ppwwyyxx) [inline detectron2/data/detection_utils.py:582]:
```suggestion
    if is_train and cfg.INPUT.RANDOM_FLIP != "none":
```

File: detectron2/data/detection_utils.py [hunk 4]
```diff
@@ -579,8 +579,13 @@ def build_augmentation(cfg, is_train):
...
+        augmentation.append(
+            T.RandomFlip(
+                horizontal=True if "horizontal" in cfg.INPUT.RANDOM_FLIP else False,
+                vertical=True if "vertical" in cfg.INPUT.RANDOM_FLIP else False,
```
Comment (ppwwyyxx) [inline detectron2/data/detection_utils.py:586]:
```
horizontal=cfg.INPUT.RANDOM_FLIP == "horizontal",
vertical=...
```

File: detectron2/data/detection_utils.py (hunk 3 above)
Comment (kondela) [reply detectron2/data/detection_utils.py]: Oops, that was left by accident from the previous commit. Should be correct now. :)

File: detectron2/data/detection_utils.py (hunk 4 above)
Comment (kondela) [reply detectron2/data/detection_utils.py]: Same as https://github.com/facebookresearch/detectron2/pull/1976#discussion_r483426629 - accidentally left from the previous commit, should be correct now.

**Pull request closed without merging**